pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -840,6 +841,14 @@ func (l *LibvirtDomainManager) generateConverterContext(vmi *v1.VirtualMachineIn
 	return c, nil
 }
 
+func computeFreePageReporting(vmi *v1.VirtualMachineInstance) bool {
+	if vmi.IsHighPerformanceVMI() || (vmi.GetAnnotations() != nil && vmi.GetAnnotations()[v1.FreePageReportingDisabledAnnotation] == "true") {
+		return false
+	}
+
+	return true
+}
+
```

Comment:
supernit

```suggestion
func computeFreePageReporting(vmi *v1.VirtualMachineInstance) bool {
	return !vmi.IsHighPerformanceVMI() && (vmi.GetAnnotations() == nil || vmi.GetAnnotations()[v1.FreePageReportingDisabledAnnotation] != "false")
}
```

Reply:
@fossedihelm:

We should consider `autoattachMemBalloon` as well - freePageReporting shouldn't be enabled when `autoattachMemBalloon` is disabled.

Reply:
`GetAnnotations` returns map[string]string
https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta.GetAnnotations

Reply:
I would use another name for computeFreePageReporting - if it returns a boolean, it should answer a question. what do you think?

Reply:
> We should consider `autoattachMemBalloon` as well - freePageReporting shouldn't be enabled when `autoattachMemBalloon` is disabled.

Actually, `autoattachMemBalloon` is considered here https://github.com/kubevirt/kubevirt/blob/8579f816ed0a02b45ef11eef7a13d923746a4870/pkg/virt-launcher/virtwrap/converter/converter.go#L1134:L1148 
but I agree that for completeness it is better to consider it also here.
 

> `GetAnnotations` returns map[string]string https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta.GetAnnotations

Right! Thank you!


> I would use another name for computeFreePageReporting - if it returns a boolean, it should answer a question. what do you think?

I opted for `isFreePageReportingEnabled`. WDYT?

Thanks